packages/core/src/client.test.ts

```diff
@@ -2385,13 +2385,61 @@ describe('Client', () => {
     callback({ key: 'randomKey' });
     expect(mockReload).not.toHaveBeenCalled();
 
+    const practitioner1 = randomUUID();
+    const practitioner2 = randomUUID();
+
+    // Should NOT refresh when we have the same profile
+    mockReload.mockReset();
+    callback({
+      key: 'activeLogin',
+      oldValue: JSON.stringify({ profile: { resourceType: 'Practitioner', id: practitioner1 } }),
+      newValue: JSON.stringify({ profile: { resourceType: 'Practitioner', id: practitioner1 } }),
+    } as StorageEvent);
+    expect(mockReload).not.toHaveBeenCalled();
+
+    // Should refresh when we change to a new profile
     mockReload.mockReset();
-    callback({ key: 'activeLogin' });
+    callback({
+      key: 'activeLogin',
+      oldValue: JSON.stringify({ profile: { resourceType: 'Practitioner', id: practitioner1 } }),
+      newValue: JSON.stringify({ profile: { resourceType: 'Practitioner', id: practitioner2 } }),
+    } as StorageEvent);
     expect(mockReload).toHaveBeenCalled();
 
+    // Should refresh when going to a new profile from no profile
     mockReload.mockReset();
-    callback({ key: null });
+    callback({
+      key: 'activeLogin',
+      oldValue: JSON.stringify({ profile: { resourceType: 'Practitioner', id: null } }),
+      newValue: JSON.stringify({ profile: { resourceType: 'Practitioner', id: practitioner1 } }),
+    } as StorageEvent);
+    expect(mockReload).toHaveBeenCalled();
+
+    // Should refresh when going from a profile to no profile (logged out)
+    mockReload.mockReset();
+    callback({
+      key: 'activeLogin',
+      oldValue: JSON.stringify({ profile: { resourceType: 'Practitioner', id: practitioner1 } }),
+      newValue: JSON.stringify({ profile: { resourceType: 'Practitioner', id: null } }),
+    } as StorageEvent);
     expect(mockReload).toHaveBeenCalled();
+
+    // Garbled values should result in a logged error and not throw
+    const originalConsoleError = console.error;
+    console.error = jest.fn();
+    mockReload.mockReset();
+    callback({
+      key: 'activeLogin',
+      oldValue: "{ profile: 'bla' ", // Invalid JSON
+      newValue: JSON.stringify({ profile: { resourceType: 'Practitioner', id: null } }),
+    } as StorageEvent);
+    expect(mockReload).not.toHaveBeenCalled();
+    expect(console.error).toHaveBeenCalled();
+    console.error = originalConsoleError;
+
+    mockReload.mockReset();
+    callback({ key: null });
```

Comment:
@codyebberson is there some specific reason why changing the behavior at this line might be bad? Sending `null` currently will no longer trigger the reload. I can change the logic if we still need it to trigger it, I just wasn't sure when `null` might be passed as the key to a storage event in our case

Reply:
Talked offline: we want it to trigger reload since `null` key means storage is being cleared.

Reply:
As discussed offline, `key: null` is the event for `localStorage.clear()`, and yeah, I think we want to preserve that case.